crates/net/network/src/transactions/mod.rs

```diff
@@ -1102,21 +1108,46 @@ where
                     metrics_increment_egress_peer_channel_full,
                 );
             }
-            // drain commands
+
+            // advance commands (propagate/fetch/serve txns)
             if let Poll::Ready(Some(cmd)) = this.command_rx.poll_next_unpin(cx) {
                 this.on_command(cmd);
                 some_ready = true;
             }
 
-            // drain incoming transaction events
+            // advance incoming transaction events (stream new txns/announcements from network
+            // manager and queue for import to pool/fetch txns)
+            //
+            // can potentially queue around 1,5k txns *
+            // `DEFAULT_BUDGET_TRY_DRAIN_NETWORK_TRANSACTION_EVENTS` = 1,5k * 1024 txns for import
+            // to pool if txns are valid. each message could contain up to soft limit
+            // byte size for response / small legacy tx size: 128 KiB / 100 bytes < 1,5k
+            // transactions.
+            //
+            // if txns however are invalid, and just 1 byte, since this isn't validated until
+            // import to pool, this can potentially queue around 0,5 billion txns. more if the
+            // message size is bigger than 128 KiB.
+            //
+            // this will potentially remove hashes from hashes pending fetch (if same hashes are
+            // announced that didn't fit into a previous request)
             if let Poll::Ready(Some(event)) = this.transaction_events.poll_next_unpin(cx) {
                 this.on_network_tx_event(event);
                 some_ready = true;
             }
 
             this.update_fetch_metrics();
 
-            // drain fetching transaction events
+            // advance fetching transaction events (flush transaction fetcher and queue for
+            // import to pool)
+            //
+            // can potentially queue around 20k txns * `DEFAULT_BUDGET_TRY_DRAIN_STREAM` = 20k *
+            // 1024 txns for import to pool if txns are valid. each message could
```

Comment:
is queuing for import inserting into the pool?

Reply:
what?